.github/workflows/ci.yml

```diff
@@ -126,11 +127,44 @@ jobs:
           LICENSE_SERVER: ${{ secrets.LICENSE_SERVER }}
           GH_USERNAME: ${{ secrets.GH_USERNAME }}
           GH_PAT: ${{ secrets.REPO_DOWNLOAD_PAT }}
-          MAPDL_IMAGE: '${{ env.DOCKER_PACKAGE }}:v21.2.1'
-  
-      - name: Build Documentation
+          MAPDL_IMAGE: '${{ env.DOCKER_PACKAGE }}:${{ env.DOCKER_IMAGE_VERSION_DOCS_BUILD }}'
+
+      - name: Retrieve PyMAPDL version
+        run: |
+          echo "::set-output name=PYMAPDL_VERSION::$(python -c "from ansys.mapdl.core import __version__; print(__version__)")"
+          echo "PyMAPDL version is: $(python -c "from ansys.mapdl.core import __version__; print(__version__)")"
+        id: version
+
+      - name: Cache examples
+        uses: actions/cache@v2
+        with:
+          path: doc/source/examples
+          key: Examples-${{ steps.version.outputs.PYMAPDL_VERSION }}-${{ github.sha }}
+          restore-keys: |
+            Examples-${{ steps.version.outputs.PYMAPDL_VERSION }}
+
+      - name: Cache docs build directory
+        uses: actions/cache@v2
+        with:
+          path: doc/build
+          key: doc-build-${{ steps.version.outputs.PYMAPDL_VERSION }}-${{ github.sha }}
```

Comment:
Won't this mean that this will be different each time?

Reply:
yes... but...

since the "restore-key" is used as a glob pattern, it will match the last version of the cache. I need to "create" a different sha every time, otherwise you cannot store the cache if there is another workflow run going on.

I hope this makes sense.